radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -46,6 +47,138 @@ pub struct TransactionReceiptV1 {
     pub resources_usage: Option<ResourcesUsage>,
 }
 
+#[cfg(feature = "std")]
+impl TransactionReceiptV1 {
+    pub fn generate_execution_breakdown_flamegraph(
+        &self,
+        path: impl AsRef<std::path::Path>,
+        title: impl AsRef<str>,
+    ) -> Result<(), FlamegraphError> {
+        let path = path.as_ref();
+        let title = title.as_ref().to_owned();
+
+        // The options to use when constructing the flamechart.
+        let mut opts = inferno::flamegraph::Options::default();
+        "Execution Cost Units".clone_into(&mut opts.count_name);
+        opts.title = title;
+
+        // Transforming the detailed execution cost breakdown into a string understood by the flamegraph
+        // library.
+        let Some(ref cost_breakdown) = self.fee_details else {
+            return Err(FlamegraphError::CostBreakdownNotAvailable);
+        };
+
+        let flamegraph_string = Self::transform_detailed_execution_breakdown_into_flamegraph_string(
+            &cost_breakdown.detailed_execution_cost_breakdown,
+        );
+
+        // Writing the flamegraph string to a temporary file since its required by the flamegraph lib to
+        // have a path.
+        let result = {
+            let tempdir = tempfile::tempdir().map_err(FlamegraphError::IOError)?;
+            let tempfile = tempdir.path().join("file.txt");
```

Comment:
True, I think that the main issue is that there isn't quite an easy way to get a file path from a `std::fs::File` and the flamegraph library primarily works with paths.

I guess a sufficiently large random number might also be useable here instead of `file.txt`? Or perhaps even better we could just hash the `detailed_execution_cost_breakdown` and make that the name of the temporary file.

Reply:
A quick google of "rust tempfile crate get path" gives me this https://docs.rs/tempfile/latest/tempfile/struct.TempPath.html - does that work?

Reply:
Ah this is great! Will use that then!